x-pack/plugins/reporting/server/native_controller_client/synthetic_process.js

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+
+
+import { Readable } from 'stream';
+
+class SyntheticStream extends Readable {
+  constructor(options) {
+    super(options);
+  }
+
+  _read() {
+  }
+}
+
+export class SyntheticProcess {
+
+  _callbacks = {};
+
+  killed = false;
+  stdout = new SyntheticStream();
```

Comment:
This and the `stderr` stream are never ended, which is probably okay because we're currently killing the process if the native controller exits, but seems risky to me.

Reply:
We're subscribing to them via the subscriptions so when the process dies, we'll stop listening here. I could close them when we kill the process, if you think that's safer.

Reply:
I was thinking of them more as being piped somewhere, but if they're being consumed by observables maybe we want them to be Subjects rather than streams... Either way, probably has no material effect